analyzer/codechecker_analyzer/analyzer.py

```diff
@@ -237,6 +253,20 @@ def perform_analysis(args, skip_handler, context, actions, metadata_tool,
 
         metadata_tool['analyzers'][analyzer] = metadata_info
 
+    if 'makefile' in args and args.makefile:
```

Comment:
this can be misleading if the perform_analysis function does not do the analysis but generates a makefile, could this be done before the perform_analysis somewhere?

Reply:
That would be a bigger change (it uses variables, parameters from above) and I don't want to do this in this pull request. It can lead to multiple bugs.

Reply:
I'm a bit worried that it will not be refactored any time soon.